app/src/main/java/com/example/arc_exapmle/MainActivity.kt

```diff
@@ -50,7 +50,22 @@ class MainActivity : AppCompatActivity() {
 
         noteViewModel.getAllNotes().observe(
             this,
-            Observer { t -> adapter.setNote(t) }
+            Observer { t: List<NoteEntity>? ->
+
+                val noteUIList: MutableList<NoteUI> = ArrayList()
+
+
+
+                for (i in t!!.indices) {
+
+                    noteUIList.add(NoteUI(t[i]))
```

Comment:
behtare toye ViewModel
NoteEntity ro be NoteUI tabdil koni baad be in livedata bedi
intori faghat settesh mikoni roye adapter.

Reply:
in ham belakhare hal shod .